file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java

```diff
@@ -232,42 +250,51 @@ private void directoryHandler(FileObject[] children) throws FileServerConnectorE
         // Sort the files
         String strSortParam = fileProperties.get(Constants.FILE_SORT_PARAM);
         if (strSortParam != null && !"NONE".equals(strSortParam)) {
-            log.debug("Starting to sort the files in folder: " + FileTransportUtils.maskURLPassword(fileURI));
-            String strSortOrder = fileProperties.get(Constants.FILE_SORT_ORDER);
-            boolean bSortOrderAscending = true;
-            if (strSortOrder != null) {
-                try {
-                    bSortOrderAscending = Boolean.parseBoolean(strSortOrder);
-                } catch (RuntimeException re) {
-                    log.warn(Constants.FILE_SORT_ORDER + " parameter should be either " +
-                            "\"true\" or \"false\". Found: " + strSortOrder +
-                            ". Assigning default value \"true\".", re);
+            if (!(Constants.FILE_SORT_VALUE_NAME.equals(strSortParam) ||
+                    Constants.FILE_SORT_VALUE_SIZE.equals(strSortParam) ||
+                    Constants.FILE_SORT_VALUE_LASTMODIFIEDTIMESTAMP.equals(strSortParam))) {
+                log.warn("Invalid value given for " + Constants.FILE_SORT_PARAM + " parameter. "
+                        + " Expected one of the values: " + Constants.FILE_SORT_VALUE_NAME + ", "
+                        + Constants.FILE_SORT_VALUE_SIZE + " or "
+                        + Constants.FILE_SORT_VALUE_LASTMODIFIEDTIMESTAMP + ". Found: " + strSortParam);
+            } else {
+                log.debug("Starting to sort the files in folder: " + FileTransportUtils.maskURLPassword(fileURI));
+                String strSortOrder = fileProperties.get(Constants.FILE_SORT_ORDER);
+                boolean bSortOrderAscending = true;
+                if (strSortOrder != null) {
+                    try {
+                        bSortOrderAscending = Boolean.parseBoolean(strSortOrder);
+                    } catch (RuntimeException re) {
+                        log.warn(Constants.FILE_SORT_ORDER + " parameter should be either " +
+                                "\"true\" or \"false\". Found: " + strSortOrder +
+                                ". Assigning default value \"true\".", re);
+                    }
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Sorting the files by : " + strSortOrder + ". (" +
+                            bSortOrderAscending + ")");
+                }
+                if (strSortParam.equals(Constants.FILE_SORT_VALUE_NAME) && bSortOrderAscending) {
```

Comment:
+1. Done in https://github.com/wso2/carbon-transports/pull/228/commits/bd704ce1531f38c52ffc2b20070992a82323887d